src/com/lid/intellij/translateme/actions/TranslateAction.java

```diff
@@ -12,114 +12,130 @@
+		ConfigurationState state = PersistingService.getInstance().getState();
+		return state.isAutoDetect();
+	}
+
+	public static boolean isTranslationTooltip() {
+		ConfigurationState state = PersistingService.getInstance().getState();
+		return state.isTranslationTooltip();
+	}
+
+	private static class PopupActionHandler implements ActionHandler {
+		@Override
+		public void handleResult(Editor editor, List<String> translated) {
+			Application app = ApplicationManager.getApplication();
+			app.invokeLater(() -> {
+				if (isTranslationTooltip()) {
+					StringBuilder sb = new StringBuilder();
+					int iMax = translated.size();
+					String sep = "";
+					for (int i = 0; i < iMax; i++) {
+						sb.append(sep).append(translated.get(i));
+						sep = "\n";
+					}
+					sb.append("<br><p><a href='http://translate.yandex.com/'>Powered by Yandex.Translator</a>");
+					showTooltip(editor, sb.toString());
+				} else {
+					ResultDialog resultDialog = new ResultDialog("Translated", translated);
+					resultDialog.setVisible(true);
+				}
+			});
+		}
+
+		@Override
+		public void handleError(Editor editor) {
+			Application app = ApplicationManager.getApplication();
+			app.invokeLater(() -> {
+				showErrorBalloon(editor, "Failed to translate");
+			});
+		}
+
+		private void showErrorBalloon(Editor editor, String message) {
+			NOTIFICATION_GROUP.createNotification("TranslateMe Error", XmlStringUtil.wrapInHtml(message), NotificationType.ERROR, (notification, event) -> {
+			}).notify(null);
+		}
+
+		static int getVisibleAreaWidth(Editor editor) {
+			Insets insets = editor.getContentComponent().getInsets();
+			int width = Math.max(0, editor.getScrollingModel().getVisibleArea().width - insets.left - insets.right);
+			return width;
+		}
+
+		private void showTooltip(Editor editor, String message) {
+			//String text = "<table><tr><td>&nbsp;</td><td>$message</td><td>&nbsp;</td></tr></table>"
+			HintManagerImpl hintManager = (HintManagerImpl) HintManager.getInstance();
+			int flags = HintManager.HIDE_BY_CARET_MOVE | HintManager.UPDATE_BY_SCROLLING | HintManager.HIDE_BY_ESCAPE;
+			int timeout = 60000; // default 1min?
+			String html = HintUtil.prepareHintText(message, HintUtil.getInformationHint());
+
+			JComponent label = HintUtil.createInformationLabel(html, e -> {
+				if (e.getEventType() == ACTIVATED) {
+					BrowserUtil.browse("http://translate.yandex.com/");
+				}
+			}, null, null);
+
+			LightweightHint hint = new LightweightHint(label);
+
+			VisualPosition start = editor.offsetToVisualPosition(editor.getSelectionModel().getSelectionStart());
+			VisualPosition end = editor.offsetToVisualPosition(editor.getSelectionModel().getSelectionEnd());
+
+			CharSequence chars = editor.getDocument().getCharsSequence();
+			int i = editor.getSelectionModel().getSelectionStart();
+			int iMax = editor.getSelectionModel().getSelectionEnd();
+			int maxColumn = Math.max(start.column, end.column);
+			for (; i < iMax; ++i) {
```

Comment:
java 8 stream api?

Reply:
You have `<idea-version since-build="131"/>` which is IntelliJ IDEA 13 https://www.jetbrains.org/intellij/sdk/docs/basics/getting_started/build_number_ranges.html#branch-and-build-numbers-for-intellij-platform-pre-20162

Older IDE versions require Java 1.6 byte code in the plugins.